File: JAVA/sort012.java
class HelloWorld {
    public static void main(String[] args) {
         int[] array = {0, 1, 1, 0, 1, 2, 1, 2, 0, 0, 0, 1};
        int[] result = sortZeroOnesTwos(array);
        
        for(int i = 0; i < result.length; i++){
            System.out.println(result[i]);
        }
    }
       public static int[] sortZeroOnesTwos(int[] arr){
        int start = 0;
        int mid = 0;
        int end = arr.length - 1;

        while(mid <= end){
            if(arr[mid] == 0){
                int temp = arr[mid];
                arr[mid] = arr[start];
                arr[start] = temp;
                mid++;
                start++;
            }else if(arr[mid] == 1){
                mid++;
            }else {
                int temp = arr[mid];
                arr[mid] = arr[end];
                arr[end] = temp;
                end--;
            }
        }
        return arr;
    }
}

File: sort012.js
// Online Javascript Editor for free
// Write, Edit and Run your Javascript code using JS Online Compiler

function sort3(arr) {
  left = 0;
  mid = 0;
  right = arr.length - 1;
  while (mid <= right) {
    if (arr[mid] === 0) {
      let temp = arr[mid];
      arr[mid] = arr[left];
      arr[left] = temp;
      left++;
      mid++;
    } else if (arr[mid] === 1) {
      mid++;
    } else {
      let temp = arr[right];
      arr[right] = arr[mid];
      arr[mid] = temp;
      right--;
    }
  }

  return arr;
}

console.log(sort3([0, 1, 1, 0, 1, 2, 1, 2, 0, 0, 0, 1]));
